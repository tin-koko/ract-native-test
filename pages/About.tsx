import React from "react";
import { Image, ScrollView, StyleSheet, Text, View, SafeAreaView } from 'react-native';

export default function About() {
  return (
    <SafeAreaView>
      <Image
        style={styles.tinyLogo}
        source={require('../assets/tinkoko.png')}
      />
      <ScrollView style={styles.scrollview}>
        <Text style={styles.name}>
          Tinkoko is a social agro e-commerce company based in Nigeria. The
          Company is Pioneered to bridge the gap between Farmers and customers,
          assist the buying and Selling of Agricultural Products and Services
          while providing a customer centric platform for perfect consumer
          satisfaction.
        </Text>
        <Text style={styles.name}>
          Tinkoko aims at providing an online market place for Farmers,
          Agri-preneurs, Agri-investors, and Consumers to meet and transact on a
          trusted & reliable Platform
        </Text>
        <Text style={styles.name}>
          For many years producers of agricultural raw materials and finished
          products have found it difficult to communicate with the consumers of
          their product to be able to get feedbacks on the products they produce
          and know what to improve on, also finding it difficult to sell their
          produce easily. The era of internet has helped in the evolution of
          different platform to bring people together for a particular purpose.
        </Text>
        <Text style={styles.name}>
          Tinkoko.com is an online platform that brings agricultural producers,
          farmers and consumers together to advertise products, negotiate
          products and get reviews and feedback on how to improve their
          products.
        </Text>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  name: {
    fontSize: 16,
    padding: 10,
  },
  tinyLogo: {
    width: 80,
    height: 80,
  },
  scrollview: {
    marginBottom: 80
  },
  flag: {
    fontSize: 100,
  },
  view: {
    fontWeight: "700",
    backgroundColor: "tomato",
    fontSize: 20,
    padding: 30,
    textAlign: "center",
  },
});
