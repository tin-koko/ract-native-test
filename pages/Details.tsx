import React, { Fragment } from "react";
import { StyleSheet, Text, View } from "react-native";

export default function Details({ route }: any) {
  const data = route?.params?.params;
  interface languageTypes {
    code: string;
    name: string;
    native: string;
  }
  return (
    <View>
      {data && (
        <View style={styles.view}>
          <Text style={styles.flag}>{data?.emoji}</Text>
          <Text style={styles.label}>Country:</Text>
          <Text style={styles.name}>
            {data.name}({data?.code})
          </Text>
          <Text style={styles.label}>Currency:</Text>
          <Text style={styles.name}>{data.currency}</Text>
          <Text style={styles.label}>Phone code:</Text>
          <Text style={styles.name}>{data.phone}</Text>         
          {data?.languages?.map((x: languageTypes) => (
            <Fragment key={x.code}>
              <Text style={styles.label}>Language:</Text>
              <Text style={styles.name}>{x?.name}</Text>
              <Text style={styles.label}>Native:</Text>
              <Text style={styles.name}>{x?.native}</Text>
            </Fragment>
          ))}
          <Text style={styles.label}>Continent:</Text>
          <Text style={styles.name}>
            {data?.continent?.name}({data?.continent?.code})
          </Text>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  name: {
    fontSize: 20,
    marginBottom: 10
  },
  label: {
    fontSize: 14,
    color: "#666666",
    lineHeight: 17,
    marginBottom: 2
  },
  flag: {
    fontSize: 180,
    textAlign: "center"
  },
  view: {
    fontWeight: "700",
    backgroundColor: "#fbfbfb",
    fontSize: 20,
    padding: 30,
    textAlign: "center",
  },
});
