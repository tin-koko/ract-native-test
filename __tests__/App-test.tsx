import React from 'react';
import About from '../pages/About';
import renderer from 'react-test-renderer';
import Details from '../pages/Details';


describe('About', () => {
it('renders correctly the about page', () => {
  const tree = renderer
    .create(<About/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly the more details page when a country in the list is clicked', () => {
  const tree = renderer
    .create(<Details/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

})


