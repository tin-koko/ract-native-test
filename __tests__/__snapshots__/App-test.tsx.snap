// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly the about page 1`] = `
<RCTSafeAreaView>
  <Image
    source={
      {
        "testUri": "../../../assets/tinkoko.png",
      }
    }
    style={
      {
        "height": 80,
        "width": 80,
      }
    }
  />
  <RCTScrollView
    style={
      {
        "marginBottom": 80,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko is a social agro e-commerce company based in Nigeria. The Company is Pioneered to bridge the gap between Farmers and customers, assist the buying and Selling of Agricultural Products and Services while providing a customer centric platform for perfect consumer satisfaction.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko aims at providing an online market place for Farmers, Agri-preneurs, Agri-investors, and Consumers to meet and transact on a trusted & reliable Platform
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        For many years producers of agricultural raw materials and finished products have found it difficult to communicate with the consumers of their product to be able to get feedbacks on the products they produce and know what to improve on, also finding it difficult to sell their produce easily. The era of internet has helped in the evolution of different platform to bring people together for a particular purpose.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko.com is an online platform that brings agricultural producers, farmers and consumers together to advertise products, negotiate products and get reviews and feedback on how to improve their products.
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`About renders correctly the more details page when a country in the list is clicked 1`] = `<View />`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView>
  <Image
    source={
      {
        "testUri": "../../../assets/tinkoko.png",
      }
    }
    style={
      {
        "height": 80,
        "width": 80,
      }
    }
  />
  <RCTScrollView
    style={
      {
        "marginBottom": 80,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko is a social agro e-commerce company based in Nigeria. The Company is Pioneered to bridge the gap between Farmers and customers, assist the buying and Selling of Agricultural Products and Services while providing a customer centric platform for perfect consumer satisfaction.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko aims at providing an online market place for Farmers, Agri-preneurs, Agri-investors, and Consumers to meet and transact on a trusted & reliable Platform
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        For many years producers of agricultural raw materials and finished products have found it difficult to communicate with the consumers of their product to be able to get feedbacks on the products they produce and know what to improve on, also finding it difficult to sell their produce easily. The era of internet has helped in the evolution of different platform to bring people together for a particular purpose.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko.com is an online platform that brings agricultural producers, farmers and consumers together to advertise products, negotiate products and get reviews and feedback on how to improve their products.
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`renders correctly the about page 1`] = `
<RCTSafeAreaView>
  <Image
    source={
      {
        "testUri": "../../../assets/tinkoko.png",
      }
    }
    style={
      {
        "height": 80,
        "width": 80,
      }
    }
  />
  <RCTScrollView
    style={
      {
        "marginBottom": 80,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko is a social agro e-commerce company based in Nigeria. The Company is Pioneered to bridge the gap between Farmers and customers, assist the buying and Selling of Agricultural Products and Services while providing a customer centric platform for perfect consumer satisfaction.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko aims at providing an online market place for Farmers, Agri-preneurs, Agri-investors, and Consumers to meet and transact on a trusted & reliable Platform
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        For many years producers of agricultural raw materials and finished products have found it difficult to communicate with the consumers of their product to be able to get feedbacks on the products they produce and know what to improve on, also finding it difficult to sell their produce easily. The era of internet has helped in the evolution of different platform to bring people together for a particular purpose.
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "padding": 10,
          }
        }
      >
        Tinkoko.com is an online platform that brings agricultural producers, farmers and consumers together to advertise products, negotiate products and get reviews and feedback on how to improve their products.
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
